xc/xc7/techmap/fix_carry.py

```diff
@@ -0,0 +1,380 @@
+""" Script for addressing CARRY4 output congestion in elaborated netlists.
```

Comment:
Should this really be a Yosys pass eventually?

Reply:
No.  The only reason we have this is because VPR cannot reliably resolve CO/O congestion.  In many/most cases, the packer should grab registered outputs from the carry chain, and this logic isn't required at all.  In the case where the carry outputs are not, or cannot be registered together, a transparent latch should be emitted, but again VPR doesn't have support for transparent latches.

I'd argue this is a reasonable band-aid for VPR until we make VPR's packer able to handle these cases.

Reply:
The other option would be to emit an absorb-able transparent latch, but this probably hairy and the current state of the VPR packer would likely do poorly with that.

Reply:
Fair enough.